File: Leetcode July Challenge/C++/Custom Sort String.cpp
class Solution {
public:
    string customSortString(string order, string str) {
        
        string s1, s2;
        int a[26]={0};
        for(char c: order) a[c-'a']++;
        for(char c: str){
            if(a[c-'a']==0) s2+=c;
            else a[c-'a']++;
        }
        for(char c: order){
            while(a[c-'a']>1){
                s1+=c; 
                a[c-'a']--;
            } 
        }
        return s1+s2;
    }
};

File: Leetcode July Challenge/Python/Custom Sort String.py
from collections import defaultdict
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        arr = [0] * 27
        print(arr)
        for i in range(len(order)):
            arr[ord(order[i]) - ord('a')] += 1
        
        s1 = "" 
        s2 = ""
        print(ord(order[0]) - ord('a'))
        for i in range(len(s)):
            if arr[ord(s[i]) - ord('a')] == 0:
                s2 += s[i]
            else:
                arr[ord(s[i]) - ord('a')]+=1
                
        for i in range(len(order)):
            while arr[ord(order[i]) - ord('a')] > 1:
                s1 += order[i]
                arr[ord(order[i]) - ord('a')] -= 1
        
        return s1+s2
                
        
        
            
            
        
